packages/core/src/lib/waku_peer_exchange/peer_discovery.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  PeerDiscovery,
+  PeerDiscoveryEvents,
+  symbol,
+} from "@libp2p/interface-peer-discovery";
+import { PeerInfo } from "@libp2p/interface-peer-info";
+import { EventEmitter } from "@libp2p/interfaces/dist/src/events";
+import debug from "debug";
+
+import { multiaddrsToPeerInfo } from "../multiaddr_to_peer_info";
+
+import { WakuPeerExchange } from ".";
+
+const log = debug("waku:peer-discovery-dns");
+
+export class PeerExchangeDiscovery
+  extends EventEmitter<PeerDiscoveryEvents>
+  implements PeerDiscovery
+{
+  private _started: boolean;
+
+  constructor(public peerExchange: WakuPeerExchange, public numPeers: number) {
+    super();
+    this._started = false;
+  }
+
+  /**
+   * Start discovery process
+   */
+  async start(): Promise<void> {
+    log("Starting peer discovery via Peer Exchange");
+
+    this._started = true;
+
+    const response = await this.peerExchange.query({
+      numPeers: BigInt(this.numPeers),
+    });
+
+    response.peerInfos.forEach((peerInfo) => {
+      if (!peerInfo.ENR) return;
+
+      const _peerInfos = multiaddrsToPeerInfo(peerInfo.ENR.getFullMultiaddrs());
+
+      _peerInfos.forEach((peerInfo) => {
+        if (!this._started) return;
+        this.dispatchEvent(
+          new CustomEvent<PeerInfo>("peer", { detail: peerInfo })
```

Comment:
if you check `@libp2p/bootstrap`, it _tags_ the peers 

https://github.com/libp2p/js-libp2p-bootstrap/blob/3a519f796e36dcef416ea105150158e65457f866/src/index.ts#L136

I would suggest to do the asme but with label `peer-exchange` so we can apply logic on top of it later.